### ISSUE TITLE
Django 1.10.1 + Wagtail 1.6.2 + django-debug-toolbar 1.5 = KeyError: 'wsgi.multiprocess'

### ISSUE THREAD

**Issue by soar** [torchbox/wagtail#2989]
I'm using Django 1.10.1 + Wagtail 1.6.2 + django-debug-toolbar 1.5. And **when I click "Preview" button** in admin interface, I get following error:

```
[ERROR] [django.request] [exception/handle_uncaught_exception] Internal Server Error: /cmspnl/pages/53/edit/preview/
Traceback (most recent call last):
  File "virtualenv\lib\site-packages\django\core\handlers\base.py", line 131, in get_response
    response = middleware_method(request, response)
  File "virtualenv\lib\site-packages\debug_toolbar\middleware.py", line 122, in process_response
    bits[-2] += toolbar.render_toolbar()
  File "virtualenv\lib\site-packages\debug_toolbar\toolbar.py", line 60, in render_toolbar
    if not self.should_render_panels():
  File "virtualenv\lib\site-packages\debug_toolbar\toolbar.py", line 81, in should_render_panels
    render_panels = self.request.META['wsgi.multiprocess']
KeyError: 'wsgi.multiprocess'
```

On other pages (which are also build in Wagtail, but accessed not via preview form) - all works fine. Also on Django 1.9 + Wagtail 1.5 + django-debug-toolbar 1.5 = all works fine too.

I think this issue is Wagtail-related.

I found this problem is related to code:

```python
    def should_render_panels(self):
        render_panels = self.config['RENDER_PANELS']
        if render_panels is None:
            render_panels = self.request.META['wsgi.multiprocess']
        return render_panels
```

I've tried set `RENDER_PANELS` - but after this two Django-Toolbar panels are shown.

**Comment by SalahAdDin** [comment]
Hi man, have you problems with `django-debug-toolbar`?

I make this [issue](https://github.com/jazzband/django-debug-toolbar/issues/875) in `debug-toolbar` thinking that problem is these package, but, you put the problem here.

**Comment by soar** [comment]
@SalahAdDin toolbar works on site pages and show all information that I need. Problem is related to "Preview" mode only.

**Comment by aaugustin** [comment]
Yes, that's because the fake HTTP request wagtail uses to generate the preview doesn't respect the WSGI spec.

**Comment by tomdyson** [comment]
Thanks for the report, @soar, and for the diagnosis, @aaugustin. Can you give us any hints on what Wagtail's fake HTTP request needs to do differently?

**Comment by aaugustin** [comment]
I'm looking at this code:
https://github.com/torchbox/wagtail/blob/3b4c24e0b59638fb695cfd0f2637b8e5bc595214/wagtail/wagtailcore/models.py#L1209-L1216

It should contain all values specified as mandatory by the WSGI spec:
https://www.python.org/dev/peps/pep-3333/#environ-variables

**Comment by SalahAdDin** [comment]
@aaugustin Does your diagnosis explain my problem?

**Comment by aaugustin** [comment]
I think it does. I've hit the same problem myself (but haven't gotten around to reporting it yet) and I'm the author of the code that triggers this issue in the debug toolbar.

**Comment by mevinbabuc** [comment]
This is the setting that works for me:

```
    def show_toolbar(request):
        request.META["wsgi.multithread"] = True
        request.META["wsgi.multiprocess"] = True
        excluded_urls = ['/pages/preview/', '/pages/preview_loading/', '/edit/preview/']
        excluded = any(request.path.endswith(url) for url in excluded_urls)
        return DEBUG and not excluded

    DEBUG_TOOLBAR_CONFIG = {"SHOW_TOOLBAR_CALLBACK": show_toolbar, }
```

With this only one debug toolbar is shown in the page, as the other one is excluded.

**Comment by m1kola** [comment]
Fixed in #3002

Thanks all!
